src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -388,19 +430,37 @@ int CRouterRoutino::calcRoute(const QPointF& p1, const QPointF& p2, QPolygonF& c
 
     try
     {
+        QVariantMap map = comboDatabase->currentData(Qt::UserRole).toMap();
+        Routino_Database * data = (Routino_Database*)(map["db"].toULongLong());
+        /*
         Routino_Database * data = (Routino_Database*)comboDatabase->currentData(Qt::UserRole).toULongLong();
+        */
         if(nullptr == data)
         {
             throw QString();
         }
 
+        // load prefix profile if it is new!
+        int res = 0;
+        QString prefixpath = map["prefixpath"].toString();
+        if( currentPrefix != prefixpath)
+        {
+            currentPrefix = prefixpath;
+            res = Routino_ParseXMLProfiles(prefixpath.toStdString().data());
+            if(res)
+            {
+                QMessageBox::critical(this, "Routino...", xlateRoutinoError(Routino_errno), QMessageBox::Abort);
+                return 0;
+            }
+        }
+
```

Comment:
This code is duplicated. It should go into a private method used in both locations.

Reply:
I have done this.